src/PowerShellEditorServices/Analysis/AnalysisService.cs

```diff
@@ -333,43 +346,61 @@ public void Dispose()
             }
         }
 
-        private void EnumeratePSScriptAnalyzerCmdlets()
+        /// <summary>
+        /// Log the features available from the PSScriptAnalyzer module that has been imported
+        /// for use with the AnalysisService.
+        /// </summary>
+        private void LogAvailablePssaFeatures()
         {
-            if (_hasScriptAnalyzerModule)
+            // Save ourselves some work here
+            var featureLogLevel = LogLevel.Verbose;
+            if (_logger.MinimumConfiguredLogLevel > featureLogLevel)
             {
-                var sb = new StringBuilder();
-                var commands = InvokePowerShell(
-                "Get-Command",
-                new Dictionary<string, object>
-                {
-                    {"Module", "PSScriptAnalyzer"}
-                });
+                return;
+            }
 
-                var commandNames = commands?
-                    .Select(c => c.ImmediateBaseObject as CmdletInfo)
-                    .Where(c => c != null)
-                    .Select(c => c.Name) ?? Enumerable.Empty<string>();
+            // If we already know the module that was imported, save some work
+            if (_pssaModuleInfo == null)
+            {
+                PSObject[] modules = InvokePowerShell(
+                    "Get-Module",
+                    new Dictionary<string, object>{ {"Name", PSSA_MODULE_NAME} });
+
+                _pssaModuleInfo = modules
+                    .Select(m => m.BaseObject)
+                    .OfType<PSModuleInfo>()
+                    .FirstOrDefault();
+            }
 
-                sb.AppendLine("The following cmdlets are available in the imported PSScriptAnalyzer module:");
-                sb.AppendLine(String.Join(Environment.NewLine, commandNames.Select(s => "    " + s)));
-                this._logger.Write(LogLevel.Verbose, sb.ToString());
+            if (_pssaModuleInfo == null)
+            {
+                throw new Exception("Unable to find loaded PSScriptAnalyzer module for logging");
```

Comment:
Will this fail gracefully? We don't want all of PSES to crash if it can't find PSSA.

Reply:
This could be a Error log instead of throwing.

Reply:
Good catch. The invocation of `AnalysisService.Create()` is already wrapped in a `try/catch`, and I would prefer not to do both. Do you have a preference on catching all exceptions outside vs inside?

Reply:
Ah turns out this is already handled -- `Create` catches all exceptions, logs them as exceptions and returns `null`.